packages/core/src/utils/StorageUtil.ts

```diff
@@ -62,5 +63,13 @@ export const StorageUtil = {
     }
 
     return []
+  },
+
+  setImageUrl(imageUrl: string) {
```

Comment:
As per above, add more expressive function names + try/catch blocks as with all other functions that operate on `localStorage`. Some weird edge cases on android devices exist where localStorage is not available or dapp's block it entirely for security reasons.